joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/FileUtil.scala

```diff
@@ -1,8 +1,124 @@
 package io.joern.x2cpg.utils
 
+import java.io.{IOException, File as JFile}
+import java.nio.file.{FileAlreadyExistsException, Files, Path, SimpleFileVisitor}
 import better.files.File
 
+import java.nio.file.attribute.BasicFileAttributes
+
 object FileUtil {
+  def newTemporaryDirectory(prefix: String = "", parent: Option[JFile] = None): JFile = {
+    parent match {
+      case Some(dir) => Files.createTempDirectory(dir.toPath, prefix).toFile
+      case _         => Files.createTempDirectory(prefix).toFile
+    }
+  }
```

Comment:
We do not need this:
a) It is never used.
b) People can directly use `Files.createTempDirectory`.